src/providers/mod.rs

```diff
@@ -131,7 +136,16 @@ fn write_ssh_keys(user: User, ssh_keys: Vec<PublicKey>) -> Result<()> {
                 format!("failed to write to file {:?}", temp_file.path().display())
             })?;
         }
-
+        let name = user.name().to_string_lossy();
+        let path = file_path.to_string_lossy();
+        let messge = format!("wrote ssh authorized keys file for user:{}", name.as_ref());
+        let map = maplit::hashmap! {
+            "AFTERBURN_USER_NAME" => name.as_ref(),
+            "AFTERBURN_PATH" => path.as_ref(),
+            "MESSAGE_ID" => AFTERBURN_SSH_AUTHORIZED_KEYS_MESSAGEID,
+        };
+        logging::journal_send(logging::Priority::Info, &messge, map.iter())
+            .chain_err(|| format!("failed to send journald entry:{}", path.as_ref()))?;
```

Comment:
I don't think we want to fail if we can't write to the journal, just logging a warning is enough.

Also, this logic should be moved further below, after we have written and renamed the file.

You can even move this to its own function, and re-use it in the two (legacy and non-legacy) cases.